src/libmeasurement_kit/libevent/dns_impl.hpp

```diff
@@ -53,37 +54,66 @@ class QueryContext : public NonMovable, public NonCopyable {
     }
 };
 
+struct evdns_base_deleter {
+    void operator()(evdns_base* p) {
+        constexpr int fail_requests = 1;
+        evdns_base_free(p, fail_requests);
+    }
+};
+
+struct evaddrinfo_deleter {
+    void operator()(evutil_addrinfo* p) {
+        evutil_freeaddrinfo(p);
+    }
+};
+
+using evdns_base_uptr = std::unique_ptr<evdns_base, evdns_base_deleter>;
+using evaddrinfo_uptr = std::unique_ptr<evutil_addrinfo, evaddrinfo_deleter>;
```

Comment:
@darkk, I see that the regress tests fail. This is because `create_evdns_base()` is a template that allows to mock `evdns_base_free()` and the regress tests mock it and make sure that, for each error path, the function is called.

The mocking itself is implemented via `MK_MOCK` in `<measurement_kit/common/mock.hpp>`.

I suggest to pass the deleter as an explicit lambda to the `std::unique_ptr` such that the `evdns_base_free` that is picked up inside `create_evdns_base()` could be mocked in the unit test.

I'd also check the coverage and, if it's reduced after this commit, it would be nice to add more tests.